### PR TITLE
Use Github Workflow for CI

### PR DESCRIPTION
**Pull request by geo2a**
This PR adds the workflow descriptions to build and test KAVM.

Tasks:

- [x] Set-up the workflow that performs the same actions that the old Jenkins CI 
- [x] Remove `Jenkinsfile`
- [x] Exclude the repository from Jenkins CI

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          cd deps/k
+          git fetch --no-recurse-submodules origin 'refs/tags/*:refs/tags/*'
+          K_VERSION="$(git tag --points-at HEAD | cut --characters=2- | head -n 1)"
+          cd ../..
+
+          KAVM_COMMIT="$(git rev-parse --short=7 HEAD)"
+
+          docker build . --file Dockerfile --build-arg K_VERSION=${K_VERSION} --tag runtimeverification/avm-semantics-ci:${KAVM_COMMIT} --build-arg USER_ID=$(id -u) --build-arg GROUP_ID=$(id -g)
+
+          docker run --name avm-semantics-ci --rm -it --detach --workdir /opt/workspace --user $(id -u):$(id -g) -v ${HOME}:${HOME} -v "$(pwd):/opt/workspace" -v "/etc/passwd:/etc/passwd:ro" -v "/etc/group:/etc/group:ro" runtimeverification/avm-semantics-ci:${KAVM_COMMIT}
```
Comment (ehildenb) [inline .github/workflows/test.yml:32]: @F-WRunTime are we not supposed to do `--rm` now? In favor of the explicit `docker stop ...` at the end?

Comment (F-WRunTime) [reply .github/workflows/test.yml:32]: We should use the --rm in favor of calling `docker rm` 
The --rm flag will cause clean up automatically when `docker stop` is called.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        run: |
+          docker exec -t avm-semantics-ci /bin/bash -c 'make -j4 test-avm-semantics-prove'
+      - name: 'Test AVM semantics --- algod integration'
+        run: |
+          docker exec -t avm-semantics-ci /bin/bash -c 'make -j4 test-kavm-algod'
+
+      - name: 'Tear down Docker'
+        if: always()
+        run: |
+          docker stop avm-semantics-ci
```
[the remark this answers is not in the document]
Comment (ehildenb) [reply .github/workflows/test.yml:55]: So it should be `docker stop --time 0 avm-semantics-ci` then?

**Review by ehildenb — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (F-WRunTime) [reply .github/workflows/test.yml:55]: Yes `docker stop --time=0` Looks like the extended flag requires two dashes.

**Pull request merged**